rectangle.js

```diff
@@ -1,58 +1,37 @@
-var Rectangle = function(p){
-    var rectangle={};
-    var program = p;
-    var vao;
-    var vaoAttrib;
```

Comment:
those variables are mostly private variables so you should not attach those variables to "this" which makes them accessible from outside of the class(you should check whether it is true).

Reply:
As I understand, in ES6 implementation of classes there is no such thing as private variables (yet). So there is nothing I can do here